lib/chef/policy_builder/policyfile.rb

```diff
@@ -51,7 +51,17 @@ class UnsupportedFeature < StandardError; end
 
       class PolicyfileError < StandardError; end
 
-      RunListExpansionIsh = Struct.new(:recipes, :roles)
+      RunListExpansionIsh = Struct.new(:recipes, :roles) do
+        def to_hash_for_json
+          expanded_run_list = recipes.map { |r|
+            { type: "recipe", name: r, skipped: false, version: nil }
```

Comment:
can you add a longish comment here about all the stuff we talked about on slack? Pointer to `RunListExpansion#to_hash`, why we can set skipped to false and version to `nil`, etc.